src/provider/model/AppDirectory.ts

```diff
@@ -1,25 +1,53 @@
-import {injectable} from 'inversify';
+import {injectable, inject} from 'inversify';
+import {MaskWatch} from 'openfin-service-config/Watch';
+import {Scope} from 'openfin-service-config/Types';
+import {ScopedConfig} from 'openfin-service-config';
 
+import {Inject} from '../common/Injectables';
 import {Application, AppName} from '../../client/directory';
 import {AppIntent} from '../../client/main';
+import {AsyncInit} from '../controller/AsyncInit';
+import {ConfigurationObject} from '../../../gen/provider/config/fdc3-config';
+
+import {ConfigStore} from './ConfigStore';
 
 enum StorageKeys {
     URL = 'fdc3@url',
     APPLICATIONS = 'fdc3@applications'
 }
 
 // Demo development app directory url
-const devUrl = 'http://localhost:3923/provider/sample-app-directory.json';
-
+export const DEV_APP_DIRECTORY_URL = 'http://localhost:3923/provider/sample-app-directory.json';
 
 @injectable()
-export class AppDirectory {
+export class AppDirectory extends AsyncInit {
+    private readonly _configStore: ConfigStore;
     private _directory: Application[] = [];
     private _url!: string;
 
-    public constructor() {
-        // Set default values
-        this._url = devUrl;
+    public constructor(@inject(Inject.CONFIG_STORE) configStore: ConfigStore) {
+        super();
+
+        this._configStore = configStore;
+    }
+
+    protected async init(): Promise<void> {
+        await this._configStore.initialized;
+
+        if (process.env.NODE_ENV === 'development') {
+            // Set the application directory to our local copy during development.  In production it will be an empty string.
+            this._configStore.config.add({level: 'desktop'}, {applicationDirectory: DEV_APP_DIRECTORY_URL});
+        }
+
+        this.updateURL(this._configStore.config.query({level: 'desktop'}).applicationDirectory);
```

Comment:
Change to `updateUrl()`? Think we're using that style more often

Reply:
done